src/compute-client/src/protocol/command.rs

```diff
@@ -477,10 +478,23 @@ impl RustType<ProtoComputeParameters> for ComputeParameters {
     }
 }
 
-/// Peek at an arrangement.
+/// Metadata specific to the peek variant.
+#[derive(Arbitrary, Clone, Debug, PartialEq, Serialize, Deserialize)]
+pub enum PeekTarget {
+    /// This peek is against an index. Since this should be held in memory on
+    /// the target cluster, no additional coordinates are necessary.
+    Index,
+    /// This peek is against a Persist collection.
+    Persist {
+        /// The identifying metadata of the Persist shard.
+        metadata: CollectionMetadata,
+    },
+}
```

Comment:
Have you considered moving the `GlobalId` into this enum as well, i.e.:

```rust
pub enum PeekTarget {
    Index { id: GlobalId },
    Persist { id: GlobalId, metadata: CollectionMetadata },
}
```

That would make `PeekTarget` an actual target specification, rather than just target metadata and might be a little clearer? We could define a `collection_id` method to make it less of a hassle to get the ID only when it's needed.

Reply:
I've pushed this up as a commit. I don't have a strong preference, so I'm happy to defer to y'all on this one!